hardhat.config.js/networks.js

```diff
@@ -5,6 +5,7 @@ const networks = {
     allowUnlimitedContractSize: true,
     blockGasLimit: 30e6,
     gas: 30e6,
+    chainId: process.env.FORK_CHAIN_ID ? Number(process.env.FORK_CHAIN_ID) : 31337
```

Comment:
using cowswap involves the chain id when composing the order signature.

To make it exactly like mainnet, you'd need a chain ID of 1.